packages/vue-cli-plugin-typescript/generator/template/src/components/Facet.vue

```diff
@@ -13,33 +13,47 @@
     </div>
   </div>
 </template>
-<script>
+<script lang="ts">
+import Vue from 'vue';
 import {buildFacet} from '@coveo/headless';
-import FacetValue from './FacetValue';
+import type {
+  Facet,
+  FacetState,
+  FacetValue as HeadlessFacetValue,
+} from '@coveo/headless';
+import FacetValue from './FacetValue.vue';
 
-export default {
+export interface IFacet {
+  state: FacetState;
+  facet: Facet;
+}
+
+export default Vue.extend({
   name: 'Facet',
-  props: ['field', 'title'],
+  props: {
+    field: String,
+    title: String,
+  },
   components: {FacetValue},
   data: function () {
     return {
       state: {},
-    };
+    } as IFacet;
   },
   methods: {
-    onToggle: function (facetValue) {
+    onToggle: function (facetValue: HeadlessFacetValue) {
       this.facet.toggleSelect(facetValue);
     },
   },
   created: function () {
     this.facet = buildFacet(this.engine, {
-      options: {field: this.field, title: this.title},
```

Comment:
this one was a bug.
Thanks to the type def, I got an error saying that title was not part of the option interface